code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -1377,14 +1379,20 @@ func (sa *StorageAllocation) replaceBlobber(blobberID string, sc *StorageSmartCo
 						blobberIsKilled = true
 
 						var cp *challengePool
-						cp, e = sc.getChallengePool(sa.ID, balances)
-						if e != nil {
-							e = fmt.Errorf("could not get challenge pool of alloc: %s, err: %v", sa.ID, e)
-						}
-
-						e = sa.moveFromChallengePool(cp, d.ChallengePoolIntegralValue)
-						if e != nil {
-							e = fmt.Errorf("failed to move challenge pool back to write pool: %v", e)
+						demeterActError := cstate.WithActivation(balances, "demeter", func() (e error) { return },
```

Comment:
Why is before function empty? We need to add older changes in before function as it is and add new changes in after function.